builder/builder/builder.py

```diff
@@ -241,16 +242,218 @@ def ConstructSnakefileConfig(self) -> dict:
             }
         return c
 
+    def PackageModule_Local(self, build_path: str, node: Node) -> None:
+        """Packages a local module into the build directory
+
+        Assumes node.snakemake is a string representing the path to the module
+        """
+        # Identify local folder structure
+        if not isinstance(node.snakefile, str):
+            raise ValueError("Local module configuration expected")
+        m_path = os.path.dirname(os.path.normpath(node.snakefile))
+        try:
+            m_pathlist = m_path.split(os.path.sep)
+            m_workflow_foldername = m_pathlist[-1]
+            m_modulename_foldername = m_pathlist[-2]
+            m_type_foldername = m_pathlist[-3]
+            m_project_foldername = m_pathlist[-4]
+            m_workflows_foldername = m_pathlist[-5]
+            if m_workflows_foldername != "workflows":
+                raise IndexError
+            m_repo_name = m_pathlist[-6]
```

Comment:
Maybe this could be a dataclass? with from/to paths

Reply:
I'm actually going to leave this because the Local and Remote options are subtly different, which complicates a common dataclass implementation; instead, I have deconstructed the paths inline, i.e. `*root, second_level, top_level = [...]` which should reduce the chance of errors creeping in through misaligned indices.